src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -108,60 +106,94 @@ public override int Execute()
+                    var installedWorkloads = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(_sdkFeatureBand);
+                    var previouslyInstalledWorkloads = installedWorkloads.Intersect(workloadIds);
+                    if (previouslyInstalledWorkloads.Any())
                     {
-                        InstallWorkloads(
-                            workloadIds,
-                            _skipManifestUpdate,
-                            _includePreviews,
-                            offlineCache);
+                        Reporter.WriteLine(string.Format(LocalizableStrings.WorkloadAlreadyInstalled, string.Join(" ", previouslyInstalledWorkloads)).Yellow());
                     }
-                    else
+                    workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                    workloadIds = WriteSDKInstallRecordsForVSWorkloads(workloadIds);
+
+                    if (!_skipManifestUpdate)
                     {
-                        RunInNewTransaction(context =>
+                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) &&
+                            InstallStateContents.FromPath(installStateFilePath) is InstallStateContents installState &&
+                            (installState.Manifests != null || installState.WorkloadVersion != null))
                         {
-                            if (!TryHandleWorkloadUpdateFromVersion(context, offlineCache, out var manifests))
+                            //  If the workload version is pinned in the install state, then we don't want to automatically update workloads when a workload is installed
+                            //  To update to a new version, the user would need to run "dotnet workload update"
+                            _skipManifestUpdate = true;
+                        }
+                    }
+
+                    RunInNewTransaction(context =>
+                    {
+                       if (!_skipManifestUpdate)
+                       {
+                            if (Verbosity != VerbosityOptions.quiet && Verbosity != VerbosityOptions.q)
                             {
-                                return;
+                                Reporter.WriteLine(LocalizableStrings.CheckForUpdatedWorkloadManifests);
                             }
-                            InstallWorkloadsWithInstallRecord(context, _workloadInstaller, workloadIds, _sdkFeatureBand, manifests, offlineCache, false);
-                        });
+                            UpdateWorkloadManifests(context, offlineCache);
+                        }
+                        _workloadInstaller.InstallWorkloads(workloadIds, _sdkFeatureBand, context, offlineCache);
```

Comment:
Don't we need to refresh workload manifests at some point between here and GC?

Reply:
The `UpdateWorkloadManifests` method does call `_workloadResolver.RefreshWorkloadManifests()`.  If manifests aren't updated / installed and we're just installing a workload, then we don't need to refresh the manifests.

Reply:
Makes sense, thanks